tactics/proof_global.ml

```diff
@@ -141,17 +141,21 @@ let private_poly_univs =
   fun () -> !b
 
 (* XXX: This is still separate from close_proof below due to drop_pt in the STM *)
-let return_proof { proof } =
-  let Proof.{name=pid;entry} = Proof.data proof in
+(* XXX: Unsafe_typ:true is needed by vio files.  *)
```

Comment:
Please clarify what you mean and why it was not needed before.
BTW, the types of delayed proofs cannot contain unresolved evars.

Reply:
@gares see the discussion above, this was introduced in bf0499bc507d5a39c3d5e3bf1f69191339270729 ; why is necessary I don't know, this PR just makes this hack more explicit, but doesn't alter the semantics there.

Reply:
@mattam82 I tried to read the commit but I can't really understand. One thing is not normalizing/substituting universes, another thing is not substituting evars. You seem to put the if-then-else around both operations. Was it intended.
I don't get the why the universe thing would make sense, but not substituting evars surely does not. I guess it works because some other piece of code calls nf_evars before...

Reply:
Yeah, would be nice to understand what's going on here and maybe clean the code a bit; I tried but couldn't get the whole picture.

Meanwhile this PR preserves the old behavior so maybe you could lift the request for changes and open a separate issue?